File: 0494-target-sum/0494-target-sum.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        dp = [[-1 for i in range(sum(nums)*2 + 1)] for i in range(len(nums))]
        def rec_func(index,s):
            if index == len(nums):
                if s == target:
                    return 1
                else:
                    return 0
            
            if dp[index][s]!=-1:
                return dp[index][s]
            
            pos = rec_func(index+1, s + nums[index])
            neg = rec_func(index+1, s + (-1 * nums[index]))
            
            dp[index][s] = pos+neg
            
            return dp[index][s]
        
        return rec_func(0,0)

File: 0494-target-sum/0494-target-sum.go
func findTargetSumWays(nums []int, target int) int {
    return RecFunc(0, 0, &nums, target)
}

func RecFunc (index int, total int, numsP *[]int, target int) int {
    if index == len(*numsP){
        if total == target {
            return 1
        }else{
            return 0
        }
    }
    
    pos := RecFunc(index+1, total+ (*numsP)[index], numsP, target)
    neg := RecFunc(index+1, total+ (-1 *(*numsP)[index]), numsP, target)
    
    return pos + neg
}